OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using OWML.Common;
+using UnityEngine;
+
+namespace OWML.ModHelper.Input
+{
+    public class ModInputHandler : IModInputHandler
+    {
+        internal static ModInputHandler Instance { get; private set; }
+
+        private HashSet<IModInputCombination> _singlePressed = new HashSet<IModInputCombination>();
+        private Dictionary<long, IModInputCombination> _comboRegistry = new Dictionary<long, IModInputCombination>();
+        private HashSet<InputCommand> _gameBindingRegistry = new HashSet<InputCommand>();
+        private float[] _timeout = new float[_maxUsefulKey];
+        private int[] _gameBindingCounter = new int[_maxUsefulKey];
+        private ModInputCombination _lastPressed;
+        private float _lastUpdate;
+        private float _lastSingleUpdate;
+        private static readonly float _updateCooldown = 0.01f; //aka little bit less than one frame
+        private static readonly float _cooldown = 0.05f;
+        private static readonly float _tapKeep = 0.3f;
+        private static readonly float _tapDuration = 0.1f;
+        private static readonly int _minUsefulKey = 8;
+        private static readonly int _maxUsefulKey = 350;
+        private static readonly int _maxComboLength = 7;
+        private static readonly string xboxPrefix = "xbox_";
```

Comment:
These can be const instead of static readonly. Example:
`private const float UpdateCooldown = 0.01f;`

Reply:
And put consts above the other variables (separate with newline)

Reply:
done

Reply:
can't turn Vector2 into const though :(